scr/test/java/com/globant/ShoppingCartTest.java

```diff
@@ -0,0 +1,79 @@
+package com.globant;
+
+import static org.junit.Assert.assertEquals;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import com.globant.models.Product;
+import com.globant.services.ShoppingCartService;
+import com.globant.services.ShoppingCartServiceImpl;
+
+public class ShoppingCartTest {
+
+	private Product productMock;
+
+	@Before
+	public void setUp() throws Exception {
+		productMock = Mockito.mock(Product.class);
+	}
+
+	@Test
+	public void whenACartIsCreatedThenItIsEmpty() {
+
+		ShoppingCartService cartService = new ShoppingCartServiceImpl();
+
+		cartService.create("Yael_salim");
+
+		assertEquals(0, cartService.getCart("Yael_salim").getCartSize());
+
+	}
+
+	@Test
+	public void whenAProductIsAddedToTheCart() {
+
+		ShoppingCartService cartService = new ShoppingCartServiceImpl();
+
+		cartService.create("Yael_salim");
+
+		cartService.addProductToCart(productMock, "Yael_salim");
+
+		assertEquals(1, cartService.getCart("Yael_salim").getCartSize());
+
+	}
+
+	@Test
+	public void whenAProductIsDeletedFromTheCart() {

```

Comment:
Your TDD must be more complete, more coverage including all the services implemented and negative cases. For example whenDeleteNullProductFromCart, this case is not covered and your service will returns NullPointerException if is invoked  `cartService.deleteCart(null)`